actioncable/lib/action_cable/connection/subscriptions.rb

```diff
@@ -46,7 +46,7 @@ def remove_subscription(subscription)
       end
 
       def perform_action(data)
-        find(data).perform_action(decode_hash(data['data']))
+        find(data).perform_action ActiveSupport::JSON.decode(data['data'])
```

Comment:
I mean the fact that we are taking in the identifier and data as "Hashes / JSON objects rather than as opaque JSON-encoded strings" -- this removal effectively breaks compatibility with allowing non-backslashed JSON in, and we should be explicit in communicating this to end users.